tests/functional/p/postponed_evaluation_pep585_py39.py

```diff
@@ -0,0 +1,68 @@
+"""Test PEP 585 works as expected, starting with Python 3.9"""
+# pylint: disable=missing-docstring,unused-argument,unused-import,too-few-public-methods,invalid-name
+import collections
+import dataclasses
+import typing
+from typing import NamedTuple
+from dataclasses import dataclass
+
+
+AliasValid = list[int]
+
+class CustomIntList(typing.List[int]):
+    pass
+
+class CustomIntListError(list[int]):
+    pass
+
+cast_variable = [1, 2, 3]
+cast_variable = typing.cast(list[cast_variable], cast_variable)
+
+T = typing.TypeVar("T", list[int])
+
+(lambda x: 2)(list[int])
+
+
+# Check typing.NamedTuple
+CustomNamedTuple = typing.NamedTuple(
+    "CustomNamedTuple", [("my_var", list[int])])
+
+class CustomNamedTuple2(NamedTuple):
+    my_var: list[int]
+
+class CustomNamedTuple3(typing.NamedTuple):
+    my_var: list[int]
+
+
+# Check dataclasses
+def my_decorator(*args, **kwargs):
+    def wraps(*args, **kwargs):
+        pass
+    return wraps
+
+@dataclass
+class CustomDataClass:
+    my_var: list[int]
+
+@dataclasses.dataclass
+class CustomDataClass2:
+    my_var: list[int]
+
+@dataclass()
+class CustomDataClass3:
+    my_var: list[int]
+
+@my_decorator
+@dataclasses.dataclass
+class CustomDataClass4:
+    my_var: list[int]
+
+
+var1: set[int]
+var2: collections.OrderedDict[str, int] = {}
+
+def func(arg: list[int]):
+    pass
+
+def func2() -> list[int]:
+    pass
```

Comment:
Those new tests are on point 👍

Reply:
Thanks! I actually forgot to add the `TypedDict` ones I wanted to add 😅 Just pushed the commit: https://github.com/PyCQA/pylint/pull/4059/commits/4f2e1aa152109f2a954287942238043a81838559